src/parser.cpp

```diff
@@ -0,0 +1,30 @@
+#include "util.cpp"
+
+#include <string>
+
+namespace ada::mimesniff {
+void parse_mime_type (std::string_view input) {
+  // step 1
+  ada::mimesniff::trim_http_whitespace(input);
+
+  // step 2
+  int position = input.find_first_of('/');
+
+  // step 5
+  if (position == std::string_view::npos) {
+    return;
+  }
+
+  // step 3
+  std::string_view type = input.substr(0, position);
+
+  // step 4
+  if (type.empty() ||
+      !ada::mimesniff::contains_only_http_tokens(type)) {
+    return;
+  }
+
+  // step 6
+  position++;
```

Comment:
why do you increment this?

Reply:
`Advance position by 1. (This skips past U+002F (/).) `

Reply:
eventually we'll need position for the start of the subtype